### ISSUE TITLE
Inheritance support

### ISSUE THREAD

**Issue by sebafelis** [json-api-dotnet/JsonApiDotNetCore#598]
## Description

Entity Framework Core [Inheritance]([url](ttps://docs.microsoft.com/en-us/ef/core/modeling/inheritance)) is not fully support now. It’s support only if for each inherited type is created separeate `DBSet `in `DbContext`. But if inheritance is define by FluentAPI then we can’t use this because Resource name is presently inseparable with type property in returning JSON. 

## Solution

I suggest adding a second string parameter (_entityName_) to `ResourceAttribute `and to `AddResource `method in `ContextGraphBuilder`. It allows to separate resource name and entity type and provides support for EF Core inheritance.

## Environment

- JsonApiDotNetCore Version: v4.0.0-alpha3
- Entity Framework Core Versions: 2.2

**Comment by maurei** [comment]
Can you provide us with a code example of the involved models and dbcontext you're trying to use with JsonApiDotNetCore?

**Comment by sebafelis** [comment]
Here is an example:
```
public class Blog : Identifiable
{
        [Attr]
        public string Url { get; set; }
}
```
```
public class RssBlog : Blog
{
        [Attr]
        public string RssUrl { get; set; }
}
```
```
public class AppDbContext : DbContext
{
        public AppDbContext(DbContextOptions<AppDbContext> options)
        : base(options)
        { }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<RssBlog>()
                .HasBaseType<Blog>();
        }
        public DbSet<Blog> Blogs { get; set; }
}
```

In this case is not possible to Post a new row of type `RssBlog` by JsonApiDotNetCore.

**Comment by maurei** [comment]
Can you share your `AddJsonApi( .. )` call from your startup?

**Comment by sebafelis** [comment]
```
services.AddJsonApi(options =>
{
                options.Namespace = "api/v1";
                options.DefaultPageSize = 10;
                options.IncludeTotalRecordCount = true;
                options.BuildResourceGraph((builder) =>
                {
                    builder.AddResource<Blog>("blogs");
                    builder.AddResource<RssBlog>("blogs");
                });
}, mvcBuilder);
```

**Comment by maurei** [comment]
I have reproduced the issue and am working on it :-)

**Comment by sebafelis** [comment]
I have checked the `alpha 4` version now. It is a bit better (the type is no longer associated with the resource name), but it is still not possible to achieve the result which I want to. I have prepared an example that (I hope so) illustrates what I want to achieve. The code is available in [here](https://github.com/sebafelis/JsonApiDotNetCoreInheritanceExample).
In my project I need one controller that will support types inheriting from one basic type. This is because I know that the number of inheriting types will increase with the development of the project, but I want to avoid the API changes (by adding a new controller for a new type).
In the created example it is possible to read and delete resource object with an inherited type (eg `StringUserField`) using `UserFieldsController`, but you cannot add and update information about a given entity by `UserFieldsController` which is unintuitive.
Also, the relationship links returned by `UserController`do not work.
